src/utils/composeStores.js

```diff
@@ -3,6 +3,21 @@ import pick from '../utils/pick';
 
 export default function composeStores(stores) {
   const finalStores = pick(stores, (val) => typeof val === 'function');
+
+  const dummyAction = {};
+  const dummyStore = {};
+
+  Object.keys(finalStores).forEach(key => {
+    if (finalStores[key](dummyStore, dummyAction) !== dummyStore) {
+      const message = `Your ${key} Store must return the state given to it for any unknown actions.`;
+      if (process.env.NODE_ENV === 'production') {
```

Comment:
Let's do it the other way.
In development, we warn. In production, we skip this whole check (lines 10 to 20).